flow-typed/src/ChorusClient.js

```diff
@@ -0,0 +1,12 @@
+interface ChorusClient {
+	open(url: string): Promise<Event>,
+	close(): void,
+	connect(): void,
+	publishStep(
+		pattern: string,
+		callback: (...args: Array<string>) => number | string | void,
```

Comment:
The last think we'll need to think about is providing access to the 'Chorus Context' within the callback - so that the processing logic can read values from it, and if necessary write values back into it. There's a bit more on ChorusContext here: [](http://www.chorus-bdd.org/pages/BuiltInHandlers/ChorusContext/ChorusContext/)

It's not required for a simple scenario but something we should discuss in another iteration